src/speech_to_text.cpp

```diff
@@ -112,17 +112,20 @@ SpeechToText::SpeechToText() {
 }
 
 void SpeechToText::start_listen() {
-	if (is_running == false) {
-		is_running = true;
-		worker.start(Callable(this, StringName("run")), Thread::Priority::PRIORITY_NORMAL);
+	if (this->worker == nullptr) {
+		this->worker = memnew(Thread);
```

Comment:
This will just allocate but not free the memory. Also, not sure we should even give the stop_listen method. Is there a use case for that? Usually if you listen, you just start the thread(let's say we can start it automatically) and it just runs forever. If yo udon't feed it audio, it won't transcribe.

Reply:
I think we need to stop_ listen function, the specific use case is:

1. Voice transcription: I press the button to start recording my voice and automatically generate text. When I feel the need to end a recording, I press the stop button. 

After all, keeping the inference thread running all the time has an impact on memory usage, and secondly, it constantly listens to my voice. I don't want this to happen.

Reply:
> 这只会分配内存，但不会释放内存。

Yes, I initially tried using "free/delete/wait_to_finish" releases "this ->worker" thread, but they all trigger crashes, and I haven't found the reason. So I can only assume that Godot has automatically reclaimed this thread

Reply:
> This will just allocate but not free the memory. Also, not sure we should even give the stop_listen method. Is there a use case for that? Usually if you listen, you just start the thread(let's say we can start it automatically) and it just runs forever. If yo udon't feed it audio, it won't transcribe.

Specific examples include:
You need to send a text message to your friend. You press the start button to start transcribing your voice, which generates a paragraph of text.
You press the stop button to stop transcription, edit the generated text, and send it to a friend
1. If there is no stop button, this text will continue to increase and I will not be able to edit the text smoothly
2. I hope this thread will only start when I want to record my voice, and there is no need to start listening to my voice at other times

Reply:
I wonder if we should also make it not a singleton and individual nodes better. What if we want 2 speakers and we want for both subtitles?

Reply:
Yes, we can change it to a mode that supports multiple speakers, which requires some modifications to the code.

Multiple identical models need to be run simultaneously, each model performing inference for each speaker.

The reason why I used singletons at that time was because I considered that there were not many such examples; And will running multiple models simultaneously cause lag for most consumer grade devices?

Reply:
It might, I already started the change though, I also wanted to move the logic of doing thread thingy outside of the plugin as much as possible and using async thingy from godot also.

Reply:
https://docs.godotengine.org/en/stable/classes/class_workerthreadpool.html is pretty cool